File: project/jimmer-core/src/test/java/org/babyfish/jimmer/model/IdCardValidator.java
package org.babyfish.jimmer.model;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class IdCardValidator implements ConstraintValidator<IdCard, String> {

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return value.length() == 15 || value.length() == 18;
    }
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/model/IdCardValidator.kt
package org.babyfish.jimmer.kt.model

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class IdCardValidator : ConstraintValidator<IdCard, String> {
    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean =
        value?.let {
            it.length == 15 || it.length == 18
        } ?: true
}